src/main/java/com/gradle/enterprise/conventions/customvalueprovider/GradleEnterpriseConventions.java

```diff
@@ -42,16 +43,38 @@
 
     public GradleEnterpriseConventions(ProviderFactory providerFactory) {
         this.providerFactory = providerFactory;
-        this.gradleEnterpriseServerUrl = getSystemProperty(GRADLE_ENTERPRISE_URL_PROPERTY_NAME, PUBLIC_GRADLE_ENTERPRISE_SERVER, providerFactory);
+        this.gradleEnterpriseServerUrl = determineGradleEnterpriseUrl();
         this.isCiServer = !getEnvVariable(CI_ENV_NAME, "").isEmpty();
     }
 
-    public String customValueSearchUrl(Map<String, String> search) {
+    private String determineGradleEnterpriseUrl() {
+        Provider<String> geServerUrl = providerFactory.systemProperty(GRADLE_ENTERPRISE_URL_PROPERTY_NAME).forUseAtConfigurationTime();
+        if (geServerUrl.isPresent()) {
+            return geServerUrl.get();
+        } else if (isAuthenticatedForDefaultGEServer()) {
+            return "https://" + DEFAULT_GRADLE_ENTERPRISE_SERVER;
+        } else {
+            // So that we can publish to default GE instance (https://gradle.com)
+            return null;
+        }
+    }
+
+    private boolean isAuthenticatedForDefaultGEServer() {
```

Comment:
This is true for CI servers, though for local builds the credentials are stored somewhere else. So if we use the new version, then local builds would publish to scans.gradle.com and not ge.gradle.org, right?